mule-extensions-api-xml-dsl/src/main/java/org/mule/runtime/extension/xml/dsl/api/resolver/DslSyntaxResolver.java

```diff
@@ -223,7 +219,7 @@ public void visitObject(ObjectType objectType) {
 
         builder.withNamespace(namespace, namespaceUri)
             .withElementName(getTopLevelTypeName(objectType))
```

Comment:
If the Type is not registered as a Top level element (`@NoGlobalDeclaration`), this element should not have an `ElementName`, no?